### PR TITLE
Nms.ft.kilkaricontrollerit

### PR DESCRIPTION
**Pull request by srivastavaAbhi**
Kilkari Api test cases merged in this branch for review.

### CODE REVIEW

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java [hunk 1]
```diff
@@ -1933,4 +1932,2003 @@ public void verifyFT21() throws IOException, InterruptedException {
...
+                .getStatusCode());
+
+        // circle blank(single space)
+        httpDelete = createDeactivateSubscriptionHttpDelete("1000000000", "A",
+                " ", "123456789012345", subscriptionId);
+
+        response = SimpleHttpClient.httpRequestAndResponse(httpDelete,
+                ADMIN_USERNAME, ADMIN_PASSWORD);
+
+        assertEquals(HttpStatus.SC_OK, response.getStatusLine()
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: The test says you're testing for failure, but you're checking for success here.

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java [hunk 2]
```diff
@@ -1933,4 +1932,2003 @@ public void verifyFT21() throws IOException, InterruptedException {
...
+                123, // callDurationInPulses
+                1, // callStatus
+                1, // callDisconnectReason
+                null)); // content
+        String expectedJsonResponse = createFailureResponseJson("<callingNumber: Not Present>");
+
+        HttpResponse response = SimpleHttpClient.httpRequestAndResponse(
+                httpPost, ADMIN_USERNAME, ADMIN_PASSWORD);
+        assertEquals(HttpStatus.SC_BAD_REQUEST, response.getStatusLine()
+                .getStatusC
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: No JIRA issue link

Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: Missing JIRA issue link

**Comment by frankhuster** [comment]
Couple of comments to deal with, looking good so far!

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java (hunk 1 above)
Comment (srivastavaAbhi) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: Added

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java (hunk 2 above)
Comment (srivastavaAbhi) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: Added

Comment (srivastavaAbhi) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: Added

**Comment by srivastavaAbhi** [comment]
Also merged #348 , #347 ,#346 ,#349 in the same branch.

**Comment by srivastavaAbhi** [comment]
merged #350 and #351 in the branch.

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java (hunk 1 above)
Comment (srivastavaAbhi) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: As per New Api doc version 1.6 , this is successful scenario. 
As soon as FT doc is changed, I will update the description.

**Comment by srivastavaAbhi** [comment]
@frankhuster : Incorporated your comments.

**Pull request merged**